src/python/test/test.py

```diff
@@ -0,0 +1,192 @@
+        )
+
+        server = tritonserver.Server(server_options).start()
+
+        http_options = KServeHttp.Options()
+        http_service = KServeHttp.Server(server, http_options)
+        http_service.start()
+
+        yield server, http_service
+
+        http_service.stop()
+        server.stop()
+
+    def test_server_ready(self, server_service):
+        server, http_service = server_service
+        with Client(httpclient, url="0.0.0.0:8000") as http_client:
+            assert http_client.is_server_ready()
+
+    def test_server_live(self, server_service):
+        server, http_service = server_service
+        with Client(httpclient, url="0.0.0.0:8000") as http_client:
+            assert http_client.is_server_live()
+
+    def test_already_exists_error(self, server_service):
+        server, http_service = server_service
+        # Should throw error because http service already started.
+        with pytest.raises(AlreadyExistsError):
+            http_service.start()
+
+    def test_wrong_parameters(self, server_service):
+        server, _ = server_service
+        with pytest.raises(Exception):
+            incorrect_http_options = KServeHttp.Options(port=-15)
+
+    def test_invalid_server(self):
+        with pytest.raises(tritonserver.InvalidArgumentError):
+            invalid_server = tritonserver.Server(tritonserver.Options())
+            custom_http_service = KServeHttp.Server(
+                invalid_server, KServeHttp.Options()
+            )
+
+    def test_inference(self):
+        model_name = "identity"
+        with Client(httpclient, url="0.0.0.0:8000") as http_client:
+            input_data = np.array([["testing"]], dtype=object)
+            # Create input and output objects
+            inputs = [httpclient.InferInput("INPUT0", input_data.shape, "BYTES")]
+            outputs = [httpclient.InferRequestedOutput("OUTPUT0")]
+
+            # Set the data for the input tensor
+            inputs[0].set_data_from_numpy(input_data)
+
+            results = http_client.infer(model_name, inputs=inputs, outputs=outputs)
+
+            output_data = results.as_numpy("OUTPUT0")
+
+            assert input_data[0][0] == output_data[0][0].decode()
+
+
+class TestKServeGrpc:
+    @pytest.fixture(scope="class")
+    def server_service(self):
+        # Making empty model repository
+        model_path = "./test_model_repository"
+
+        # Starting Server Instance
+        server_options = tritonserver.Options(
+            server_id="TestServer",
+            model_repository=model_path,
+            log_error=True,
+            log_warn=True,
+            log_info=True,
+        )
+
+        server = tritonserver.Server(server_options).start()
+
+        grpc_options = KServeGrpc.Options()
+        print(grpc_options)
+        grpc_service = KServeGrpc.Server(server, grpc_options)
+        grpc_service.start()
+
+        yield server, grpc_service
+
+        grpc_service.stop()
+        server.stop()
```

Comment:
What happens if you stop server `server.stop()` before frontend `service.stop()`? Is it graceful? And should we have a test to enforce the expected behavior?

Reply:
For example,
```python
# simple ordering test
server.stop()
service.stop() # is this OK?
```

```python
# runtime behavior test
server.stop()

# send inference request
# what happens?

service.stop()
```